src/socket.c

```diff
@@ -2194,6 +2199,11 @@ int camblet_connect(struct sock *sk, struct sockaddr *uaddr, int addr_len)
 		WRITE_ONCE(sk->sk_prot, prot);
 		WRITE_ONCE(sk->sk_socket->ops, prot_ops);
 	}
+	else
```

Comment:
The same applies to line 2154. Also, I think there will be a leak, when the sc is already populated so we are not initiating it on line 2162, and an error happens in line 2182.